### ISSUE TITLE
GPG Public Key Missing

### ISSUE THREAD

**Issue by mikenye** [just-containers/s6-overlay#352]
```
$ curl https://keybase.io/justcontainers/key.asc | gpg --import
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    32  100    32    0     0     20      0  0:00:01  0:00:01 --:--:--    20
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
```

```
$ curl https://keybase.io/justcontainers/key.asc
SELF-SIGNED PUBLIC KEY NOT FOUND
```

Thanks!

**Comment by jprjr** [comment]
@glerchundi I believe you set up the keybase.io account? I'm not sure how to get in there and correct the issue.

As a work-around keys can be pulled from the MIT keyserver with something like:

```
gpg --keyserver pgp.mit.edu --recv-keys 6101B2783B2FD161
```

**Comment by regismesquita** [comment]
Just a friendly FYI , Looks like pop.mit.edu is having some reliability issues, I have some requests taking quite long and some failing (both local and on CI).
Offline copy seems as the most reliable solution right now , but can be dangerous in the long run.

**Comment by DavidSpek** [comment]
pgp.surfnet.nl is a reliable solution. It's from a Dutch organization that does the infrastructure for universities and research facilities. https://pgp.surfnet.nl/ has a nice interface as well.

**Comment by srbala** [comment]
> Just a friendly FYI , Looks like pgp.mit.edu is having some reliability issues

We have moved our script to use hkps://keyserver.ubuntu.com

**Comment by skarnet** [comment]
Can I close this?

**Comment by mikenye** [comment]
Can the documentation be updated to reflect how to verify the download given the updated keyserver?

**Comment by mikenye** [comment]
I've submitted a PR to update the README with the new commands to import the gpg public key.

IMHO, this should be left open until releases are signed by gpg again.

**Comment by jprjr** [comment]
Regarding GPG signing releases - should we continue it?

My reasoning - it's all automated. I've got the keys in the repo, encrypted, and using GitHub actions "I" decrypt them and sign the release - but it's all triggered by somebody with write access pushing a tag. At the end of the day, "I'm" not actually taking steps to authenticate. GitHub is the ultimate authenticator/authorizor - if you can write a tag to the repo, you can create a release.

If we required the keyholder to somehow authenticate that yes - they're a real person authorizing this release - then it would make sense. But it seems like we're not really providing that much value by signing the releases. If a failure occurs, it's going to be an infrastructure failure, or the key expiring, etc - and never a case of somebody signing with an incorrect key, because they're an untrusted user trying to create a release.

I suppose it verifies that the tarball hasn't been tampered with, but that could be handled with checksums and require less infrastructure. So, I'm not sure if we should continue signing releases. But I'm open to hearing if/why we should continue.

**Comment by skarnet** [comment]
I agree that a hash should be enough - the important thing to verify is that the tarball you get is exactly the tarball we provide. The rest is nerd paranoia - which is healthy, but unjustified if signing doesn't add anything security-wise.

Is there a way to tell GitHub to always take and publish the SHA256 of the tarballs they make?

**Comment by jprjr** [comment]
As far as I know, no - the GitHub-generated tarballs are generated on-the-fly. They usually produce the same binary output, but the key phrase there is "usually" - last I checked, it's not guaranteed to always be the same binary (stuff like timestamps can change, etc).

The only way to have a reliable tarball is to upload it as a release artifact.

**Comment by skarnet** [comment]
That wouldn't be a problem if they could generate a `foo.tar.gz.sha256` file along with the `foo.tar.gz` file, but I suppose that feature doesn't exist.

**Comment by mwgamble** [comment]
I'll echo what @skarnet said - as long as there's some kind of checksum provided by upstream alongside the regular release artifacts, I don't really mind if it's not a GPG signature.

**Comment by mikenye** [comment]
> I'll echo what @skarnet said - as long as there's some kind of checksum provided by upstream alongside the regular release artifacts, I don't really mind if it's not a GPG signature.

FWIW, I agree with this.

The only reason for asking this issue to be kept open until GPG was implemented, as that I just assumed gpg signing would continue (as this was the signing process used previously).

FWIW I'm happy for checksums to be provided against binary tarballs.

Cheers.

**Comment by jprjr** [comment]
I'm going to close this issue - we've set up CI to generate sha256 checksums, which are published alongside the tarballs. That should be significantly less error-prone, and provide reasonable guarantees that the tarball wasn't tampered with.

**Comment by mikenye** [comment]
Thanks so much!
